### PR TITLE
QUERY RESULT

### PR DESCRIPTION
**Pull request by tchasinga**
```
-- Create the owners table
CREATE TABLE owners (
    id SERIAL PRIMARY KEY,
    full_name VARCHAR(255),
    age INT
);
```
Screenshot (278)

```
-- Create the species table
CREATE TABLE species (
    id SERIAL PRIMARY KEY,
    name VARCHAR(255)
);
```

Screenshot (277)

```
-- Create the owners table
CREATE TABLE owners (
    id SERIAL PRIMARY KEY,
    full_name VARCHAR(255),
    age INT
);

-- Create the species table
CREATE TABLE species (
    id SERIAL PRIMARY KEY,
    name VARCHAR(255)
);
```

Screenshot (279)

```
SELECT * FROM ANIMALS
```
Screenshot (280)

```
-- Modify the animals table
ALTER TABLE animals
    ADD COLUMN species_id INT REFERENCES species(id),
    ADD COLUMN owner_id INT REFERENCES owners(id);
```

Screenshot (281)

```
-- Insert data into the owners table
INSERT INTO owners (full_name, age) VALUES
    ('Sam Smith', 34),
    ('Jennifer Orwell', 19),
    ('Bob', 45),
    ('Melody Pond', 77),
    ('Dean Winchester', 14),
    ('Jodie Whittaker', 38);
```
Screenshot (282)

```
INSERT INTO species (name) VALUES
    ('Pokemon'),
    ('Digimon');
```
Screenshot (283)

```
-- Update animals based on species
UPDATE animals SET species_id = 1 WHERE name NOT LIKE '%mon';
UPDATE animals SET species_id = 2 WHERE name LIKE '%mon';
```
Screenshot (284)

```
-- Update animals with owner information
UPDATE animals SET owner_id = 4 WHERE name = 'Charmander' OR name = 'Squirtle' OR name = 'Blossom';
UPDATE animals SET owner_id = 5 WHERE name = 'Angemon' OR name = 'Boarmon';
UPDATE animals SET owner_id = 2 WHERE name = 'Gabumon' OR name = 'Pikachu';
UPDATE animals SET owner_id = 3 WHERE name = 'Devimon' OR name = 'Plantmon';
UPDATE animals SET owner_id = 1 WHERE name = 'Agumon';
```
Screenshot (285)

```
   QUERY TO JOIN 

-- What animals belong to Melody Pond?
SELECT a.name
FROM animals a
JOIN owners o ON a.owner_id = o.id
WHERE o.full_name = 'Melody Pond';

-- List of all animals that are pokemon (their type is Pokemon).
SELECT a.name
FROM animals a
JOIN species s ON a.species_id = s.id
WHERE s.name = 'Pokemon';

-- List all owners and their animals, including those that don't own any animal.
SELECT o.full_name, COALESCE(array_agg(a.name), ARRAY[]::VARCHAR[]) AS animals_owned
FROM owners o
LEFT JOIN animals a ON o.id = a.owner_id
GROUP BY o.id;

-- How many animals are there per species?
SELECT s.name, COUNT(a.id) AS animal_count
FROM species s
LEFT JOIN animals a ON s.id = a.species_id
GROUP BY s.id, s.name;

-- List all Digimon owned by Jennifer Orwell.
SELECT a.name
FROM animals a
JOIN owners o ON a.owner_id = o.id
JOIN species s ON a.species_id = s.id
WHERE o.full_name = 'Jennifer Orwell' AND s.name = 'Digimon';

-- List all animals owned by Dean Winchester that haven't tried to escape.
SELECT a.name
FROM animals a
JOIN owners o ON a.owner_id = o.id
WHERE o.full_name = 'Dean Winchester' AND a.escape_attempts = 0;

-- Who owns the most animals?
SELECT o.full_name, COUNT(a.id) AS animal_count
FROM owners o
LEFT JOIN animals a ON o.id = a.owner_id
GROUP BY o.id, o.full_name
ORDER BY animal_count DESC
LIMIT 1;
```
Screenshot (286)

```
-- Remove the existing species column from animals table
ALTER TABLE animals DROP COLUMN species;

-- Displaying results 
 select * from animals
```

Screenshot (287)

### CODE REVIEW

**Review by AbelG101 — COMMENTED** [review]
# Hi @tchasinga 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- You have submitted a project, that doesn't follow the proper GitFlow, the merge request should be into a `dev` branch not into `main`.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by addod19** [comment]
## Hi @tchasinga, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because you did not follow the correct Git-flow workflow

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: schema.sql [hunk 1]
```diff
@@ -11,3 +11,74 @@ CREATE TABLE animals (
...
+CREATE TABLE species (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255)
+);
+
+-- Modify the animals table
+ALTER TABLE animals
+    ADD COLUMN species_id INT REFERENCES species(id),
+    ADD COLUMN owner_id INT REFERENCES owners(id);
+
```
Comment (AbelG101) [inline schema.sql]: - [ ] As part of the project requriements please make sure to add the statement for removing the `species` column from the animals table.

File: data.sql
Comment (AbelG101) [inline data.sql:1]: - [ ] Kindly make sure to add the queries for each question in the `queries.sql` file I can see that you've copied the queries in the PR description but they should be present in the specified file to give a meaningful review.
- [ ] I also noticed you're screenshot is incomplete, please make sure to add a screenshot for each query and it's respective result.

Comment (AbelG101) [inline data.sql:1]: - [ ] Kindly make sure to add the queries for each question in the `queries.sql` file I can see that you've copied the queries in the PR description but they should be present in the specified file to give a meaningful review.
- [ ] I also noticed you're screenshot is incomplete, please make sure to add a screenshot for each query and it's respective result.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @tchasinga 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Data is properly inserted ✔
- New columns are correctly added ✔
- Correct use of Git Flow ✔
- Descriptive PR title ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Jacques, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## From the note history it seems you deleted one of the comments left by the previous reviewer, Please don't do this.

## Highlights
- Good job creating the tables

## Required Changes ♻️
- [ ] it seems that there are no changes pushed to the repository after the last review, please make sure to push all the requested changes to the Repo.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tchasinga** [comment]
@mikethreels  dear hi  tell to delete that comment because he made it by defaults that's why I did by my self I'll never such of things please give me a review , and don't take it in bad side for really

**Comment by tchasinga** [comment]
@mikethreels  I've made All change that he asked me to done

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @tchasinga,

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] As the last reviewer mentioned, There are no new changes pushed yet, Please push theme because it seems that you forgot to push them.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @tchasinga, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**